Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm

```diff
@@ -1342,6 +1342,42 @@ - (void)testCanClearPersistenceAfterRestarting {
   [self awaitExpectations];
 }
 
+- (void)testCanClearPersistenceOnANewFirestoreInstance {
+  FIRDocumentReference *doc = [self documentRef];
+  FIRFirestore *firestore = doc.firestore;
+  FIRApp *app = firestore.app;
+  NSString *appName = app.name;
+  FIROptions *options = app.options;
+
+  NSDictionary<NSString *, id> *initialData = @{@"foo" : @"42"};
+  [self writeDocumentRef:doc data:initialData];
+
+  [firestore terminateWithCompletion:[self completionForExpectationWithName:@"Terminate"]];
+  [self.firestores removeObject:firestore];
+  [self awaitExpectations];
+  [self deleteApp:app];
+
+  // We restart the app with the same name and options to check that the previous instance's
+  // persistent storage is actually cleared after the restart. Calling [self firestore] here would
+  // create a new instance of firestore, which defeats the purpose of this test.
+  [FIRApp configureWithName:appName options:options];
+  FIRApp *app2 = [FIRApp appNamed:appName];
+  FIRFirestore *firestore2 = [self firestoreWithApp:app2];
+  [firestore2
```

Comment:
Note that the only material difference between this test case and the one above it is that the location of the call to clearPersistenceWithCompletion() has moved to be invoked on the newly-created firestore2 object.